plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1161,44 +1165,76 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
 
+        Set<String> unsupportedProperties = Sets.difference(properties.keySet(), UPDATABLE_PROPERTIES);
+        if (!unsupportedProperties.isEmpty()) {
+            throw new TrinoException(NOT_SUPPORTED, "The following properties cannot be updated: " + String.join(", ", unsupportedProperties));
+        }
+
         transaction = icebergTable.newTransaction();
         UpdateProperties updateProperties = transaction.updateProperties();
 
-        for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
-            String trinoPropertyName = propertyEntry.getKey();
-            Optional<Object> propertyValue = propertyEntry.getValue();
+        Optional<Object> fileFormat = properties.get(FILE_FORMAT_PROPERTY);
+        if (fileFormat != null && fileFormat.isPresent()) {
+            updateProperties.defaultFormat(((IcebergFileFormat) fileFormat.get()).toIceberg());
+        }
 
-            switch (trinoPropertyName) {
-                case FILE_FORMAT_PROPERTY:
-                    updateProperties.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg());
-                    break;
-                case FORMAT_VERSION_PROPERTY:
-                    // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
-                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
-                    break;
-                default:
-                    // TODO: Support updating partitioning https://github.com/trinodb/trino/issues/12174
-                    throw new TrinoException(NOT_SUPPORTED, "Updating the " + trinoPropertyName + " property is not supported");
-            }
+        Optional<Object> formatVersion = properties.get(FORMAT_VERSION_PROPERTY);
+        if (formatVersion != null && formatVersion.isPresent()) {
+            // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
+            updateProperties.set(FORMAT_VERSION, Integer.toString((int) formatVersion.get()));
```

Comment:
per https://github.com/trinodb/trino/pull/12259#discussion_r874619676 , i think `formatVersion=null` means "no change", but `formatVersion!=null && formatVersion.isEmpty()` means "reset to default" (which is "v2").

please double check me.

Reply:
No, looks like if the value is being reset to the default, the properties map will have the default value from IcebergTableProperties https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/metadata/PropertyUtil.java#L68

It will be Optional.empty() if that default happens to be null, but it's not for any of the properties here.

Reply:
There's a test for that in TestIcebergV2 where setting `format_version = DEFAULT` on a v1 table upgrades it to v2